src/arcsf/data/tofu.py

```diff
@@ -0,0 +1,193 @@
+import math
+
+import numpy as np
+from datasets import Dataset, load_dataset
+from sklearn.model_selection import train_test_split
+
+
+def _flatten_list_of_lists(list: list[list]) -> list:
+    return sum(list, [])
+
+
+# returns the question index in the dataset, given an author number and question
+# number (within the author) (inner function)
+def _get_forget_index(
+    author: int,
+    question: int,
+    q_per_author: int,
+) -> int:
+    return question + q_per_author * author
+
+
+# As above returns the question indices in the dataset, given an
+# author number and question numbers (within the author) (inner function)
+def _get_forget_indices(
+    authors: int,
+    questions: int | list[int],
+    q_per_author: int,
+) -> int | list[int]:
+    if isinstance(questions, int):
+        return _get_forget_index(authors, questions, q_per_author)
+    if isinstance(questions, list):
+        return [
+            _get_forget_index(authors, question, q_per_author) for question in questions
+        ]
```

Comment:
The 2nd `if` isn't really needed (if it's not an int it's required to be a list per the docstring):

```suggestion
    if isinstance(questions, int):
        return _get_forget_index(authors, questions, q_per_author)

    return [
        _get_forget_index(authors, question, q_per_author) for question in questions
    ]
```

Alternatively, if you do want to check the type it should raise an error if it's neither of those types (at the moment it would just get to the end of the function and return None)

Reply:
Changed this behaviour